client/my-sites/upgrades/checkout-thank-you/index.jsx

```diff
@@ -163,7 +163,7 @@ var CheckoutThankYou = React.createClass( {
 	 * @returns {*[]} an array of varying size with the component instance, then an optional purchase object possibly followed by a domain name
 	 */
 	getComponentAndPrimaryPurchaseAndDomain: function() {
-		if ( this.isDataLoaded() ) {
+		if ( this.isDataLoaded() && this.props.receiptId ) {
```

Comment:
Why not just update `isDataLoaded` to be?

`this.props.receiptId && this.props.receipt.hasLoadedFromServer;`

After all, if there's no receiptId, then it can't ever be loaded...